src/software/new_geom/point.h

```diff
@@ -0,0 +1,451 @@
+
+    /**
+     * Creates a new Point from a Vector
+     *
+     * @param the Vector to create a Point from
+     */
+    constexpr Point(const Vector &v);
+
+    /**
+     * Returns the x coordinate of this Point
+     *
+     * @return the x coordinate of this Point
+     */
+    constexpr double x() const;
+
+    /**
+     * Returns the y coordinate of this Point
+     *
+     * @return the y coordinate of this Point
+     */
+    constexpr double y() const;
+
+    /**
+     * Sets the coordinates of this point to the new coordinates
+     *
+     * @param x the new x coordinate
+     * @param y the new y coordinate
+     */
+    void set(double x, double y);
+
+    /**
+     * Sets the x coordinate of this point
+     *
+     * @param x the new x coordinate
+     */
+    void setX(double x);
+
+    /**
+     * Sets the y coordinate of this point
+     *
+     * @param y the new y coordinate
+     */
+    void setY(double y);
+
+    /**
+     * Returns the distance between this Point and origin (0,0)
+     *
+     * @return the distance
+     */
+    double distanceFromOrigin() const;
+
+    /**
+     * Returns the distance between this Point and the given point p
+     *
+     * @return the distance
+     */
+    double distanceFromPoint(const Point &p) const;
+
+    /**
+     * Returns a new Vector from this Point
+     *
+     * @return A new vector from this Point
+     */
+    Vector toVector() const;
+
+    /**
+     * Returns the unit vector in direction of this point
+     *
+     * @return A unit vector in the direction of this Point, or a
+     * zero-vector if this Point is zero
+     */
+    Vector norm() const;
+
+    /**
+     * Returns a scaled normalized vector in the same direction as this
+     * Point
+     *
+     * @param length the desired length of the resultant vector
+     *
+     * @return a vector in the same direction as this Point and with the given length,
+     * or a zero-vector if this Point is zero
+     */
+    Vector norm(double length) const;
+
+    /**
+     * Returns a new Point that is this Point rotated counterclockwise by an angle
+     *
+     * @param rot the angle to rotate the Point
+     *
+     * @return the new Point rotated by rot
+     */
+    Point rotate(const Angle &rot) const;
+
+    /**
+     * Returns the direction of this Point
+     *
+     * @return the direction of this Point, in the range [-π, π], with 0 being
+     * the positive x direction, π/2 being up (positive y), etc. like on a standard x-y
+     * plane
+     *
+     *              +
+     *              | Y
+     *              |
+     *              |
+     *      +---------------+
+     *       -X     |      X
+     *              |
+     *              | -Y
+     *              +
+     */
+    Angle orientation() const;
+
+    /**
+     * Checks whether this Point contains NaN in either coordinate
+     *
+     * @return true if either coordinate is NaN, and false otherwise
+     */
+    constexpr bool isnan() const;
```

Comment:
Thoughts on this function? There's currently no usages of it, so I think we can remove it. Not to say it's not useful, but we already aren't using it anywhere and generally should be avoiding cases where things become nan. Unit tests should catch any major issues.